src/net.cpp

```diff
@@ -1044,7 +1051,7 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     pnode->m_permissionFlags = permissionFlags;
     // If this flag is present, the user probably expect that RPC and QT report it as whitelisted (backward compatibility)
     pnode->m_legacyWhitelisted = legacyWhitelisted;
-    pnode->m_prefer_evict = bannedlevel > 0;
+    pnode->m_prefer_evict = banned || discouraged;
```

Comment:
What if they are `PF_NOBAN`? .. these flags may be set then and thus `m_prefer_evict` will be set... is that acceptable?

Reply:
PF_NOBAN peers are already except from eviction entirely, so the value of the `m_prefer_evict` field has no meaning for them.

The code here is just maintaining the existing semantics (current master also sets `m_prefer_evict` for (auto)banned peers with PF_NOBAN).